### ISSUE TITLE
Some issues are ignored when always_run is present

### ISSUE THREAD

**Issue by sbaudoin** [ansible-community/ansible-lint#1086]
##### Summary
This issue has appeared with ansible-lint 4.3.0. In a playbook where `always_run` is present for a task, some other issues are ignored.

##### Issue Type

- Bug Report

##### Ansible and Ansible Lint details
```console
ansible --version
ansible 2.8.16
  config file = None
  configured module search path = ['/root/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python3.7/dist-packages/ansible
  executable location = /usr/local/bin/ansible
  python version = 3.7.3 (default, Jul 25 2020, 13:03:44) [GCC 8.3.0]

ansible-lint --version
ansible-lint 4.3.5
```

- ansible installation method: pip
- ansible-lint installation method: pip

##### OS / ENVIRONMENT
Ubuntu 20.04 LTS

##### STEPS TO REPRODUCE
Run ansible-lint with the following playbook:
```yaml
- hosts: all
  tasks:
  - name: unset variable
    command: echo {{thisvariable}} is not set in this playbook
  - name: always run
    debug:
      msg: "always_run is deprecated"
    always_run: true
```

##### Desired Behaviour
The expected behaviour is the same as with ansible-lint 4.2.0 as follows with the complete playbook (i.e. with `always_run`)):
```console
example.yml:4: [E301] Commands should not change things if nothing needs doing
example.yml:5: [E206] Variables should have spaces before and after: {{ var_name }}
example.yml:6: [E101] Deprecated always_run
```

##### Actual Behaviour
If you check the playbook detailed above with ansible-lint 4.3.x (0 to 6 are impacted), here is the result:
```console
example.yml:5: [E206] Variables should have spaces before and after: {{ var_name }}
example.yml:6: [E101] Deprecated always_run
```

If you comment out (or remove) the last line with `always_run`, here is the result:
```console
example.yml:4: [E301] Commands should not change things if nothing needs doing
example.yml:5: [E206] Variables should have spaces before and after: {{ var_name }}
```

When `always_run` is present, the issue E301 is no longer returned.
